### PR TITLE
Get rid of the old build logic and remove deprecated build dependencies

### PR DESCRIPTION
**Pull request by umutuzgur**
* Build Appium via node 8
* Add the logic in shell scripts to the Jenkinsfile
* Put the node binaries into the docker image for iOS and Android
* Remove patched logic from code since we don't do that anymore
* Use a really simple base image for the data only images
* Add WDA for Appium 1.7.2

### CODE REVIEW

File: appium/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM busybox
+
+ARG APPIUM_VERSION
+
+ADD . /root/appium/appium/$APPIUM_VERSION
+
+VOLUME /root/appium/appium/$APPIUM_VERSION
+ENTRYPOINT ["/bin/echo", "foo"]
```
Comment (clample) [inline appium/Dockerfile]: Do we need this entrypoint?

Comment (umutuzgur) [reply appium/Dockerfile]: Yes. Docker enforces us to have something as a CMD or ENTRYPOINT

File: Jenkinsfile [hunk 2]
```diff
@@ -66,8 +51,39 @@ node {
...
 
     deleteDir() // don't want to waste space after a build
-    notifySlack("Build Appium Version `${tag}` successful")
+    notifySlack("Build Appium Version `${params.APPIUM_VERSION}` successful")
+}
+def copyNodeBinaries() {
+    echo "Copying node"
+    sh "mkdir -p appium/node/linux/bin"
+    sh "mkdir -p appium/node/osx/bin"
+    sh "cp node/linux/bin/node appium/node/linux/bin"
```
Comment (clample) [inline Jenkinsfile]: This copies the node version from the docker container (`8.9.4`)?

File: Jenkinsfile [hunk 3]
```diff
@@ -55,8 +40,8 @@ node {
 	stage("push") {
         try {
 			if (params.PUSH_IMAGE) {
-				sh "docker tag testobject-appium:${tag} testobject/appium:${tag}"
-    			sh "docker push testobject/appium:${tag}"
+				sh "docker tag testobject-appium:${params.APPIUM_VERSION} testobject/appium:${params.APPIUM_VERSION}"
+    			sh "docker push testobject/appium:${params.APPIUM_VERSION}"
```
Comment (clample) [inline Jenkinsfile]: indentation looks off

**Pull request merged**